app/scripts/lib/safe-reload.ts

```diff
@@ -29,13 +38,17 @@ export function getRequestSafeReload(persistenceManager: PersistenceManager) {
 
   return {
     /**
-     * Safely updates the persistence manager with the provided parameters.
+     * Safely updates the persistence manager
      *
-     * @param params - Parameters to be passed to the persistence manager's
-     * `set` method.
+     * @param params - Arguments to pass to the persistence operation. For
+     * 'data' storage, pass the state; for 'split' storage, no arguments needed.
      * @returns true if the update was queued, false if writes are not allowed.
      */
-    update: async (...params: Parameters<PersistenceManager['set']>) => {
+    persist: async (
```

Comment:
nit: clarity for downstream contexts.
```suggestion
    safePersist: async (
```

Reply:
fixed in https://github.com/MetaMask/metamask-extension/pull/37060/commits/d7991cc7041956a4d8b4251df51d673fc07f1008